src/TreeView/TreeView.tsx

```diff
@@ -0,0 +1,217 @@
+  const {hasSubTree, subTree, childrenWithoutSubTree} = useSubTree(children)
+
+  // Expand or collapse the subtree
+  const toggle = React.useCallback(() => {
+    onToggle?.(!isExpanded)
+    setIsExpanded(!isExpanded)
+  }, [isExpanded, onToggle])
+
+  return (
+    <ItemContext.Provider value={{level: level + 1, isExpanded}}>
+      <li
+        role="treeitem"
+        // TODO: aria-label for treeitem
+        aria-level={level}
+        aria-expanded={hasSubTree ? isExpanded : undefined}
+        onKeyDown={event => {
+          if (event.key === ' ' || event.key === 'Enter') {
+            onSelect?.(event)
+          }
+        }}
+      >
+        <Box
+          onClick={event => {
+            if (onSelect) {
+              onSelect(event)
+            } else {
+              toggle()
+            }
+          }}
+          sx={{
+            display: 'grid',
+            gridTemplateColumns: `calc(${level - 1} * 8px) 16px 1fr`,
+            gridTemplateAreas: `"spacer toggle content"`,
+            width: '100%',
+            height: 32,
+            fontSize: 1,
+            color: 'fg.default',
+            borderRadius: 2,
+            cursor: 'pointer',
+            transition: 'background 33.333ms linear',
+            '&:hover': {
+              backgroundColor: 'actionListItem.default.hoverBg'
+            }
+          }}
+        >
+          {hasSubTree ? (
+            <Box
+              onClick={event => {
+                if (onSelect) {
+                  toggle()
+                  event.stopPropagation()
+                }
+              }}
+              sx={{
+                gridArea: 'toggle',
+                display: 'flex',
+                alignItems: 'center',
+                justifyContent: 'center',
+                height: '100%',
+                color: 'fg.muted',
+                borderTopLeftRadius: level === 1 ? 2 : 0,
+                borderBottomLeftRadius: level === 1 ? 2 : 0,
+                '&:hover': {
+                  backgroundColor: onSelect ? 'actionListItem.default.hoverBg' : null
+                }
+              }}
+            >
+              {isExpanded ? <ChevronDownIcon /> : <ChevronRightIcon />}
+            </Box>
+          ) : null}
+          <Box
+            sx={{
+              gridArea: 'content',
+              display: 'flex',
+              alignItems: 'center',
+              height: '100%',
+              px: 2
+            }}
+          >
+            {childrenWithoutSubTree}
+          </Box>
+        </Box>
+        {subTree}
+      </li>
+    </ItemContext.Provider>
+  )
+}
+
+// ----------------------------------------------------------------------------
+// TreeView.LinkItem
+
+export type TreeViewLinkItemProps = TreeViewItemProps & {
+  href?: string
+}
+
+// TODO: Use an <a> element to enable native browser behavior like opening links in a new tab
+const LinkItem: React.FC<TreeViewLinkItemProps> = ({href, onSelect, ...props}) => {
```

Comment:
We're not using a `<a>` element here based on a recommendation from @jscholes. Later, let's investigate whether there's an accessible way to use an `<a>` element so we don't lose out on native browser link features (like opening links in a new tab).

Reply:
Oh interesting! I wonder what the accessibility concerns with `<a>` are?

Reply:
@broccolinisoup - it's not the `<a>` itself, it's the markup that would be required to make it work. It would be easier to explain in a Loom or on a Zoom call/screenshare if you want the details.